src/codegen/codegen_c_visitor.cpp

```diff
@@ -1701,6 +1707,61 @@ std::string CodegenCVisitor::find_var_unique_name(const std::string& original_na
     return unique_name;
 }
 
+bool is_functor_const(const ast::StatementBlock& variable_block,
+                      const ast::StatementBlock& functor_block) {
+    auto get_local_statement = [](const std::shared_ptr<ast::Statement>& statement) {
+        std::shared_ptr<LocalListStatement> ret_local_list_statement;
+        if (statement->is_local_list_statement()) {
+            ret_local_list_statement = std::static_pointer_cast<LocalListStatement>(statement);
+        } else {
+            auto e_statement = std::dynamic_pointer_cast<ast::ExpressionStatement>(statement);
+            auto expression = e_statement->get_expression();
+            if (expression->is_local_list_statement()) {
+                ret_local_list_statement = std::static_pointer_cast<LocalListStatement>(statement);
+            }
+        }
+        return ret_local_list_statement;
+    };
+
+    // Save DUChain for every variable in variable_block
+    std::unordered_map<std::string, DUChain> chains;
+
+    // Create complete_block with both variable declarations (done in variable_block) and solver
+    // part (done in functor_block) to be able to run the SymtabVisitor and DefUseAnalyzeVisitor
+    // then and get the proper DUChains for the variables defined in the variable_block
+    ast::StatementBlock complete_block(functor_block);
+    // Typically variable_block has only one statement, a statement containing the declaration
+    // of the local variables
+    for (const auto& statement: variable_block.get_statements()) {
+        complete_block.insert_statement(complete_block.get_statements().begin(), statement);
+    }
```

Comment:
Just as a suggestion: with the similar code, you can also create `ast::Program` and `insert` nodes into it. That will solve the problem that we were discussing.

Reply:
It's a bit pain to do this in the end. Simply adding the `StatementBlock`s to the `Program` doesn't work because then `SymtabVisitor` is crashing due to redefinition of `LOCAL` variables which are handled as `GLOBAL` being in the top level of the `Program`. Instead of this I created a new constructor for the `SymtabVisitor` to get an empty `ModelSymbolTable` and be able to process any kind of node.

Reply:
can't you just clone the variable block? Or are you filtering something out? I do not understand exactly why you do this copy. Clearly is intentional

Reply:
It's indeed optional. I am creating a new StatementBlock that just includes both the `variable` and `functor_block`s